geoportal/c2cgeoportal_geoportal/views/profile.py

```diff
@@ -48,6 +53,42 @@ class Profile(Raster):
     def __init__(self, request: pyramid.request.Request):
         Raster.__init__(self, request)
 
+    @staticmethod
+    def _to_filtered(points: list[dict[str, Any]], layers: list[str]) -> list[dict[str, Any]]:
+        profile = []
+        for point in points:
+            filtered_alts = {key: value for key, value in point["alts"].items() if key in layers}
+            profile.append(
+                {
+                    "dist": point["dist"],
+                    "values": filtered_alts,
+                    "x": point["easting"],
+                    "y": point["northing"],
+                }
+            )
+        return profile
+
+    def _get_profile_service_data(
+        self, layers: list[str], geom: dict[str, Any], rasters: dict[str, Any], nb_points: int
+    ) -> list[dict[str, Any]]:
+        request = f"{rasters[layers[0]]['url']}/profile.json?geom={geom}\
+            &nbPoints={nb_points}&distinct_points=true"
+        response = requests.get(request, timeout=10)
+        if not response.ok:
+            _LOG.warning("profile request %s failed with status code %s", request, response.status_code)
+            raise HTTPInternalServerError(
+                f"Failed to fetch profile data from {request}: \
+                {response.status_code} {response.reason}"
+            )
+
+        try:
+            points = json.loads(response.content)
+        except (TypeError, JSONDecodeError) as exc:
+            _LOG.warning("profile request %s failed", request)
+            raise HTTPInternalServerError(f"Failed to decode JSON response from {request}") from exc
```

Comment:
For me, request shouldn't be public.
```suggestion
            raise HTTPInternalServerError(f"Failed to decode JSON response frominternal request") from exc
```